source/Public/New-GuestConfigurationPolicy.ps1

```diff
@@ -275,26 +296,39 @@ function New-GuestConfigurationPolicy
 
     # Download package
     $tempPath = Reset-GCWorkerTempDirectory
+    $packagePath = Join-Path -Path $tempPath -ChildPath 'extracted'
 
-    $packageFileDownloadName = 'temp.zip'
-    $packageFileDownloadPath = Join-Path -Path $tempPath -ChildPath $packageFileDownloadName
-
-    if (Test-Path -Path $packageFileDownloadPath)
+    if ($ContentManagedIdentity -and $ContentPath)
     {
-        $null = Remove-Item -Path $packageFileDownloadPath -Force
-    }
+        $contentHash = (Get-FileHash -Path $ContentPath -Algorithm 'SHA256').Hash
 
-    $null = Invoke-WebRequest -Uri $ContentUri -OutFile $packageFileDownloadPath
+        $packagePresentIn = $ContentPath
 
-    if ($null -eq (Get-Command -Name 'Get-FileHash' -ErrorAction 'SilentlyContinue'))
+        Write-Information -MessageData "Arc is not supported for use with User Assigned Idenity" -InformationAction Continue
+    }
+    else
     {
-        $null = Import-Module -Name 'Microsoft.PowerShell.Utility'
+        $packageFileDownloadName = 'temp.zip'
+        $packageFileDownloadPath = Join-Path -Path $tempPath -ChildPath $packageFileDownloadName
+
+        if (Test-Path -Path $packageFileDownloadPath)
+        {
+            $null = Remove-Item -Path $packageFileDownloadPath -Force
+        }
+
+        $null = Invoke-WebRequest -Uri $ContentUri -OutFile $packageFileDownloadPath
```

Comment:
Since we are no longer downloading with managed identity, how are we validating that the ContentUri is valid?

Reply:
From the previous code the content uri is not getting verified for download. An error is displayed for not able to download.
This validation is kept as is - 
![image](https://github.com/Azure/GuestConfiguration/assets/141965277/a811e9bb-12b0-49c6-964c-5aa7b036bfae)

Reply:
Given that the intention of enabling this feature is to ensure that only machines with appropriate levels of access can download the package, we will extensively document this to customers to assure them that this is in fact intended behavior.